modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationRegistry.java

```diff
@@ -136,7 +136,7 @@ public void initializeConfigurationWith(ConfigurationSource configurationSource)
      * @param <T> Configuration tree type.
      * @return Public configuration tree.
      */
-    public <V, C, T extends ConfigurationTree<V, C>> T getConfiguration(RootKey<T, V> rootKey) {
+    public <V, C, T extends ConfigurationTree<? super V, C>> T getConfiguration(RootKey<T, V> rootKey) {
```

Comment:
I also doubt that this is required, as well as all the rest of `? super`, there should be an easier way